src/Chainweb/Miner/Miners.hs

```diff
@@ -96,24 +99,39 @@ submit :<|> poll = client (Proxy :: Proxy MiningAPI)
 -- on a different machine, may be on multiple machines, may be arbitrarily
 -- multithreaded.
 --
-remoteMining :: Manager -> [BaseUrl] -> BlockHeader -> IO BlockHeader
-remoteMining m urls bh = traverseConcurrently_ Par' submission urls >> polling
+-- ASSUMPTION: The contents of the given @NonEmpty BaseUrl@ are unique.
```

Comment:
Why not use a non-empty set?

Reply:
It used to, before the final version of the PR. Unfortunately `NESet` doesn't have a `Traversable` instance, which is necessary for `traverseConcurrently` below.